pkg/gitreceive/storage/interfaces.go

```diff
@@ -79,3 +79,77 @@ func (f *FakeObjectPutter) PutObject(bucketName, objectKey string, reader io.Rea
 	})
 	return f.Fn(bucketName, objectKey, reader, contentType)
 }
+
+// The s3.Object already satisfies this interface. As long as it has some or all of the functions as s3.Object, it satisfies it. We're making it have all of the functions in this case. You can create a fake object that also has all of these functions.
+type Object interface {
+	// This is called an interface composition - it automatically gives your interface the function in io.Reader (https://godoc.org/io#Reader) and the function in io.Closer (https://godoc.org/io#Closer)
+	io.ReadCloser
+	// This is also an interface composition. It gives your interface the function in io.Seeker (https://godoc.org/io#Seeker)
+	io.Seeker
+	// This is also interface composition. It gives your interface the function in io.ReaderAt (https://godoc.org/io#ReaderAt)
+	io.ReaderAt
+	// This function is the last one we have to add to make this interface have all the same functions as s3.Object
+	Stat() (s3.ObjectInfo, error)
+}
+
+// The minio client doesn't already satisfy this interface, because the GetObject func (https://godoc.org/github.com/minio/minio-go#Client.GetObject) doesn't return an Object. Instead, it returns a *s3.Object. We'll create an adapter below
+type ObjectGetter interface {
+	// GetObject is *almost* the same function as the GetObject func in the minio client, but it returns Object instead of *s3.Object
+	GetObject(string, string) (Object, error)
+}
+
+// RealObjectGetter is an adapter to make the *s3.Client GetObject function compatible with the ObjectGetter interface
+type RealObjectGetter struct {
+	Client *s3.Client
+}
+
+func (r *RealObjectGetter) GetObject(bucket, objKey string) (Object, error) {
+	obj, err := r.Client.GetObject(bucket, objKey)
+	if err != nil {
+		// you can return nil for an interface
+		return nil, err
+	}
+	// obj is a *s3.Object, but as we said above, it automatically implements our Object interface without any other work
+	return obj, nil
+}
```

Comment:
A style note, (_not_ blocking for this PR):

Since `RealObjectGetter` is essentially a pass-through implementation to the Minio client's `GetObject` call, you can most likely make it non-exported and create a constructor `func` for it that returns an `ObjectGetter` interface.

Such code would look like this:

```go
type realObjectGetter struct {
  client *s3.Client
}

func (r *realObjectGetter) GetObject(bucket, objKey string) (Object, error) {
  ...
}

func NewRealObjectGetter(mcl *s3.Client) ObjectGetter {
  return &realObjectGetter{client: mcl}
}
```